src/nvim/api/private/helpers.c

```diff
@@ -368,11 +368,10 @@ static inline void typval_encode_list_start(EncodedData *const edata,
                                             const size_t len)
   FUNC_ATTR_ALWAYS_INLINE FUNC_ATTR_NONNULL_ALL
 {
-  const Object obj = OBJECT_INIT;
   kv_push(edata->stack, ARRAY_OBJ(((Array) {
     .capacity = len,
     .size = 0,
-    .items = xmalloc(len * sizeof(*obj.data.array.items)),
+    .items = xmalloc(len * sizeof(*((Object)OBJECT_INIT).data.array.items)),
```

Comment:
That would be better, I wasn't aware you could even do that. I'll update with that change.